add some cpack configurations and rpm builds

ntorresalberto opened the pull request:
Hi,
I've added:

- Building rpm packages (with the same information that was used for .deb packages)
- The `USE_SOURCE_PERMISSIONS` allows maintaining the original permissions in the case of creation of while using `install(DIRECTORY ...)` (check [docs](https://cmake.org/cmake/help/latest/command/install.html)).
- The use of `CMAKE_INSTALL_DEFAULT_DIRECTORY_PERMISSIONS` sets default permissions. By default, it has `WORLD_WRITE` was enabled which makes no sense for installed system files.
- Using the `CPACK_DEBIAN_PACKAGE_RELEASE` (the proper way to set a build number according to [cpack docs](https://cmake.org/cmake/help/v3.6/module/CPackDeb.html)).
To expand a bit on this, `CPACK_DEBIAN_PACKAGE_VERSION` has the `MAJOR.MINOR.PATCH-RELEASE` info deduced from `CPACK_PACKAGE_VERSION` (which in term comes from the project `VERSION` string) and should not be manually modified.
Here's a quote from the [cpack docs](https://cmake.org/cmake/help/v3.6/module/CPackDeb.html):
> This is the numbering of the DEB package itself, i.e. the version of the packaging and not the version of the content (see CPACK_DEBIAN_PACKAGE_VERSION). One may change the default value if the previous packaging was buggy and/or you want to put here a fancy Linux distro specific numbering.
- With this change, future builds that *do not change the contents* (i.e. library version) but *do imply* a new release build of the same library version should be set in `PACKAGE_RELEASE` variable.
Example: you've released `libfranka-0.8.0-1` and now want to release `libfranka-0.8.0-2` (there must be a new commit with `set(PACKAGE_RELEASE 2)`).

fwalch commented on CMakeLists.txt:
```diff
@@ -187,8 +211,8 @@ if(DPKG_PROG)
   set(CPACK_SYSTEM_NAME ${CPACK_DEBIAN_PACKAGE_ARCHITECTURE})
 endif()
 # Debian versions require a dash
```
Can you remove this line?

ntorresalberto replied:
done: https://github.com/frankaemika/libfranka/pull/86/commits/54c777abca34eb829a6d2a3e7446aff6581ea68a
Though now that I think about it, according to the [cmake docs](https://cmake.org/cmake/help/v3.6/module/CPackDeb.html) `CPACK_DEBIAN_PACKAGE_ARCHITECTURE` is populated by the output of `dpkg --print-architecture` *by default*, so all that if block above seems redundant.
If you agree, I can refactor that too though I cannot test it in a debian-based distro for the moment.

fwalch replied:
Good point.. I tested it and it seems that this is indeed not required, so the block can be removed.

ntorresalberto replied:
I removed them here: https://github.com/frankaemika/libfranka/pull/86/commits/fede7423c88f4f54f208dcdf40429a788c71aa52

sgabl commented:
Thanks for the contribution!

A quick fix to fix the build:

```
diff --git a/.ci/Dockerfile.bionic b/.ci/Dockerfile.bionic
index fb92e82..07688fb 100644
--- a/.ci/Dockerfile.bionic
+++ b/.ci/Dockerfile.bionic
@@ -11,5 +11,6 @@ RUN apt-get update && apt-get install -y \
     libeigen3-dev \
     libpoco-dev \
     rename \
+    rpm \
     valgrind \
     && rm -rf /var/lib/apt/lists/*
diff --git a/.ci/Dockerfile.focal b/.ci/Dockerfile.focal
index c15aade..55c0b3b 100644
--- a/.ci/Dockerfile.focal
+++ b/.ci/Dockerfile.focal
@@ -12,5 +12,6 @@ RUN apt-get update && apt-get install -y \
     libeigen3-dev \
     libpoco-dev \
     rename \
+    rpm \
     valgrind \
     && rm -rf /var/lib/apt/lists/*
diff --git a/.ci/Dockerfile.xenial b/.ci/Dockerfile.xenial
index 0fbc135..393006e 100644
--- a/.ci/Dockerfile.xenial
+++ b/.ci/Dockerfile.xenial
@@ -11,5 +11,6 @@ RUN apt-get update && apt-get install -y \
     libeigen3-dev \
     libpoco-dev \
     rename \
+    rpm \
     valgrind \
     && rm -rf /var/lib/apt/lists/*
```

ntorresalberto commented:
> Thanks for the contribution!
> 
> A quick fix to fix the build:

I added these changes and rebased the branch over develop but github is not reflecting the changes. Should I create a new PR?
Here's the commit: https://github.com/ntorresalberto/libfranka/commit/31e5e39e6882309a020267fbcb126212268c4dac

fwalch commented:
Merged, thank you!

ntorresalberto commented:
Hi, @fwalch @sgabl 
I've recently tried to use this functionality but according to the main `CMakeLists.txt` [commit history](https://github.com/frankaemika/libfranka/commits/develop/CMakeLists.txt) these changes are no longer available since  commit cac09ea3377f76f51c5d734f4d9a69701cebc468 even though they do appear in the previous commit 700e3808e6a4289ceca0a4a1c87af0e4fca12b5f.

I don't see the removal of the relevant code in the changelog so I guess it could have been a bad merge or a push force somewhere?
Would it be possible to cherry-pick these changes?

marcbone commented:
Uh yeah something strange happened there. I am sorry. We also have the problem that some changes that happend in the CmakeList were resulting in failing builds on the ROS buildfarm. As it only happens on the buildfarm and not in the test builds for the buildfarm, we can only test by doing a full release.  So for the next time we will probably try to revert all the changes in CMakeLists.txt or maybe we find the root cause of the problem. Then we can probably reapply your changes.
